services/telemetry_v2/ds_qandb_select.go

```diff
@@ -0,0 +1,56 @@
+package telemetry_v2
+
+import (
+	"context"
+	"database/sql"
+	_ "github.com/ClickHouse/clickhouse-go/v2"
+	pmmv1 "github.com/percona-platform/saas/gen/telemetry/events/pmm"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type dsQanDbSelect struct {
+	l      *logrus.Entry
+	config DSConfigQAN
+	db     *sql.DB
+}
+
+// check interfaces
+var (
+	_ TelemetryDataSource = (*dsQanDbSelect)(nil)
+)
+
+func (d *dsQanDbSelect) Enabled() bool {
+	return d.config.Enabled
+}
+
+func NewDsQanDbSelect(config DSConfigQAN, l *logrus.Entry) (TelemetryDataSource, error) {
+	db, err := openQANDBConnection(config.DSN, config.Enabled)
+	if err != nil {
+		return nil, err
+	}
+	return &dsQanDbSelect{
+		l:      l,
+		config: config,
+		db:     db,
+	}, nil
+}
+
+func openQANDBConnection(dsn string, enabled bool) (*sql.DB, error) {
```

Comment:
let's not connect to clickhouse directly and request all required data from qan-api

Reply:
it was a requirement. we need to use separate credentials and dedicated connection